scripts/spicy/zeek.spicy

```diff
@@ -48,46 +50,82 @@ public type ProtocolHandle = __library_type("zeek::spicy::rt::ProtocolHandle");
 
 ## Adds a Zeek-side child protocol analyzer to the current connection.
 ##
-## If the same analyzer was added previously with protocol_handle_get_or_create or
-## protocol_begin with same argument, and not closed with protocol_handle_close
-## or protocol_end, no new analyzer will be added.
+## If the same analyzer was added previously with `protocol_handle_get_or_create` or
+## `protocol_begin` with same argument, and not closed with `protocol_handle_close`
+## or `protocol_end`, no new analyzer will be added.
 ##
 ## See `protocol_handle_get_or_create` for the error semantics of this function.
 ##
 ## analyzer: type of analyzer to instantiate, specified through its Zeek-side
-## name (similar to what Zeek's signature action `enable` takes); if not
-## specified, Zeek will perform its usual dynamic protocol detection to figure
-## out how to parse the data (the latter will work only for TCP protocols, though.)
-public function protocol_begin(analyzer: optional<string> = Null) : void &cxxname="zeek::spicy::rt::protocol_begin";
+## name (similar to what Zeek's signature action `enable` takes)
+##
+## protocol: the transport-layer protocol that the analyser uses; only TCP is
+## currently supported here
+##
+## Note: For backwards compatibility, the analyzer argument can be left unset to add
+## a DPD analyzer. This use is deprecated, though; use the single-argument version of
+## `protocol_begin` for that instead.
+public function protocol_begin(analyzer: optional<string>, protocol: spicy::Protocol = spicy::Protocol::TCP) : void &cxxname="zeek::spicy::rt::protocol_begin";
```

Comment:
> "use the protocol of the current analyzer"

I don't think that's a useful default: a TCP analyzer may have packet-based data to pass on to a UDP analyzer; and, similarly, a UDP analyzer could have stream-data to pass to a TCP analyzer. Just being UDP or TCP doesn't determine the next layer, whereas defaulting that way seems to suggest it usually should. 

What I'd instead prefer is actually making it mandatory the provide the protocol explicitly, instead of the TCP default right now that I put in there; that way the developer would be forced to say what it is. However, that would break backwards compability because of previous versions implicitly assuming TCP (because we don't support anything else anyways).